core/shared/src/main/scala/zio/internal/Platform.scala

```diff
@@ -77,17 +78,48 @@ trait Platform { self =>
    */
   def reportFailure(cause: Cause[Any]): Unit
 
+  /**
+   * Print Fiber Dump of all fibers when interrupt signal received
+   *
+   */
+  def withFiberDumpOnInterrupt: Platform =
```

Comment:
Is the terminology here (and other places) right? We're not dumping the fibers on *interruption*, just on receiving the signal.

Reply:
Yes its when we receive a signal. Maybe I should rename to: 

```suggestion
  def withFiberDumpOnSignal: Platform =
```

Reply:
Yes, there is a little bit of ambiguity here because we want to say we will do the fiber dump when we receive this specific `USR2` signal. So `onSignal` is a little vague and `OnInterrupt` is confusing with ZIO interruption. Potentially we could do `onInterruptSignal` or `onInterruptionSignal` or otherwise `onSignal` is probably fine and I doubt too many people are going to be adding other signal handlers.

Reply:
Yeah that'd be great :+1: There are a bunch of other places that could use the correction - on `Runtime` and on the docs

Reply:
yup I'll make the changes everywhere thanks for feedback. Agreed!